lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object_lookup.rb

```diff
@@ -0,0 +1,59 @@
+module MiqAeEngine
+  module MiqAeObjectLookup
+    def initialize_obj_entries
+      @lookup_hash = Hash.new { |h, k| h[k] = [] }
+    end
+
+    def add_obj_entry(fqns, klass, instance, object)
+      parts = fqns.split('/')
+      domain = parts.shift
+      namespace = parts.join('/')
+      key = klass.downcase
+      @lookup_hash[key] = @lookup_hash[key] << {:domain    => domain.downcase,
```

Comment:
I don't understand the assignment part of `@lookup_hash[key] = @lookup_hash[key] <<`  You can just modify it directly...

```ruby
@lookup_hash[key] << {:domain ...
```